packages/gatsby-graphiql-explorer/src/app/snippets.js

```diff
@@ -0,0 +1,59 @@
+const getQuery = arg => {
+  const { operationDataList } = arg
+  const { query } = operationDataList[0]
+  return query
+}
+
+const pageQuery = {
+  name: `Page query`,
+  language: `JavaScript`,
+  codeMirrorMode: `jsx`,
+  options: [],
+  generate: arg => `import React from "react"
+import { graphql } from "gatsby"
+
+export default ({ data }) => <pre>{JSON.stringify(data, null, 4)}</pre>
+
+export const query = graphql\`
+${getQuery(arg)}
+\`
+`,
+}
+
+const staticHook = {
+  name: `StaticQuery hook`,
+  language: `JavaScript`,
+  codeMirrorMode: `jsx`,
+  options: [],
+  generate: arg => `import React from "react"
+import { useStaticQuery, graphql } from "gatsby"
+
+export default () => {
+const data = useStaticQuery(graphql\`
+  ${getQuery(arg)}
```

Comment:
Minor formatting issue is that the first line gets spaces added, but following lines don't. Is it worth padding the lines with a couple of spaces to keep things nicely formatted?